src/hugr.rs

```diff
@@ -25,15 +25,16 @@ use pyo3::prelude::*;
 
 pub use self::views::HugrView;
 use crate::extension::ExtensionSet;
+use crate::ops::tag::TagRepr;
 use crate::ops::{OpTag, OpTrait, OpType};
 use crate::types::{FunctionType, Signature};
 
 use delegate::delegate;
-
+const ROOT_REPR: TagRepr = OpTag::ModuleRoot.repr();
 /// The Hugr data structure.
 #[derive(Clone, Debug, PartialEq)]
 #[cfg_attr(feature = "pyo3", pyclass)]
-pub struct Hugr {
+pub struct Hugr<const C: TagRepr = ROOT_REPR> {
```

Comment:
Assuming we have a custom Debug impl here (and in every other thing with tags as const params), we will see the right tag names on error messages.

The main friction point is in type-hints from `rust-analyzer`, e.g.
![image](https://github.com/CQCL-DEV/hugr/assets/121866228/29f66fc3-44e0-4b5c-b155-d568060c3a18)

I don't think there's a way around that...

Reply:
ah yeah I guess for that the char solution is better

Reply:
Hang on, what am I missing here. `C` isn't used inside the struct, or anywhere else; and is fixed (AFAICS?) at ModuleRoot. What about non-module-rooted Hugrs?

Reply:
shouldn't be fixed, I've just made ModuleRoot the default to avoid changes all over the place

Reply:
Ok, so I suspect I haven't understood this yet - it might become clearer when/if you do the "changes all over the place". But if you parameterize Hugr's (as well as views) we'll need to check that we don't change the root node's optype (via e.g. `replace_op`) in a way that invalidates the guarantee.

Hmmm, which criticism could also be levelled at the `replace_op` method of `HugrMutInternals` when used on a SiblingMut, I think. At least it's not public, oooops....